app/src/lib/git/cherry-pick.ts

```diff
@@ -41,6 +49,74 @@ export enum CherryPickResult {
   Error = 'Error',
 }
 
+/**
+ * A parser to read and emit cherry pick progress from Git `stdout`.
+ *
+ * Each successful cherry picked commit outputs a set of lines similar to the
+ * following example:
+ *    [branchName commitSha] commitSummary
+ *    Date: timestamp
+ *    1 file changed, 1 insertion(+)
+ *    create mode 100644 filename
+ */
+class GitCherryPickParser {
+  private count = 0
+  public constructor(private readonly commits: ReadonlyArray<CommitOneLine>) {}
+
+  public parse(line: string): ICherryPickProgress | null {
+    const cherryPickRe = /\[(.*\s.*)\]/
```

Comment:
Shouldn't we add the `^` character to the regex to make sure that regex is at the beginning of the line? Otherwise, maybe a file named `Hello [world].md` would trigger a match, right? 🤔 
```suggestion
    const cherryPickRe = /^\[(.*\s.*)\]/
```

Reply:
Absolutely! Thank you! I must of not copied it from my regex tester because I did have it.